kerlescan/view_helpers.py

```diff
@@ -9,6 +9,85 @@
 from kerlescan.rbac_service_interface import get_roles
 from kerlescan.exceptions import HTTPError
 
+pagination_link_template = "%s?limit=%s&offset=%s&order_by=%s&order_how=%s"
+
+
+def _create_first_link(path, limit, offset, count, order_by, order_how):
+    first_link = pagination_link_template % (path, limit, 0, order_by, order_how)
+    return first_link
+
+
+def _create_previous_link(path, limit, offset, count, order_by, order_how):
+    # if we are at the beginning, do not create a previous link
+    if offset == 0 or offset - limit < 0:
+        return _create_first_link(path, limit, offset, count, order_by, order_how)
+    previous_link = pagination_link_template % (
+        request.path,
+        limit,
+        offset - limit,
+        order_by,
+        order_how,
+    )
+    return previous_link
+
+
+def _create_next_link(path, limit, offset, count, order_by, order_how):
+    # if we are at the end, do not create a next link
+    if limit + offset >= count:
+        return _create_last_link(path, limit, offset, count, order_by, order_how)
+    next_link = pagination_link_template % (
+        request.path,
+        limit,
+        limit + offset,
+        order_by,
+        order_how,
+    )
+    return next_link
+
+
+def _create_last_link(path, limit, offset, count, order_by, order_how):
+    final_offset = count - limit if (count - limit) >= 0 else 0
+    last_link = pagination_link_template % (
+        path,
+        limit,
+        final_offset,
+        order_by,
+        order_how,
+    )
+    return last_link
+
+
+def _build_paginated_baseline_list_response(
+    count, limit, offset, order_by, order_how, baseline_list, withhold_facts=False
+):
+    json_baseline_list = [
+        baseline.to_json(withhold_facts=withhold_facts) for baseline in baseline_list
+    ]
```

Comment:
This has some baseline-specific code. You may want to pass in `json_list` instead of `baseline_list` and then compute what `json_list` is in the service code. That way the service can control exactly what it wants to send to the user.